File: Code/app/src/main/java/com/group_7/library_management/models/Book.kt
package com.group_7.library_management.models

data class Book(
    val id: String,
    val title: String,
    val author: String,
    val category: String,
    val isbn: String? = null,
    val publisher: String? = null,
    val publishYear: Int? = null,
    val totalCopies: Int = 0,
    val coverImageUrl: String? = null,
    val description: String? = null,
    val borrowFee: Long = 0L,
    val availableCopies: Int = 0,
    val rating: Double = 0.0,
    val createdAt: Long = 0,
    val ratingCount: Int = 0,
    val popularityScore: Long = 0,
)

File: Admin/src/types/Book.ts
export type BookStatus = "available" | "borrowed" | "out_of_stock";

export interface Book {
  bookId: string;
  isbn?: string;
  title: string;
  author: string;
  category: string;
  publisher?: string;
  publishYear?: number;
  quantity: number;
  availableQuantity: number;
  cover?: string;
  description?: string;
  borrowFee?: number;
  shelfLocation?: string;
  authorDetails?: BookAuthor[];
  publisherDetails?: BookPublisher;
}

export interface BookAuthor {
  id: number;
  name: string;
  penName?: string;
  birthDate?: string;
  deathDate?: string;
  nationality?: string;
}

export interface BookPublisher {
  id: number;
  name: string;
  address?: string;
  email?: string;
  phone?: string;
}

export interface BookFormData {
  isbn: string;
  title: string;
  author: string;
  category: string;
  publisher: string;
  publishYear: string;
  quantity: string;
  cover?: string;
  description: string;
  borrowFee: string;
  shelfLocation: string;
  authorDetails?: BookAuthor[];
  publisherDetails?: BookPublisher;
}

export interface BookBorrowHistory {
  id: string;
  readerId: string;
  readerName: string;
  copyBarcode: string;
  requestedDate: string;
  borrowDate: string;
  dueDate: string;
  returnDate?: string;
  status: "pending_payment" | "requested" | "borrowing" | "returned" | "overdue" | "cancelled";
}

export function getBookStatus(book: Book): BookStatus {
  if (book.availableQuantity === 0) return "out_of_stock";
  if (book.availableQuantity < book.quantity) return "borrowed";
  return "available";
}
